### PR TITLE
Standardize predict interface using SAR standard

### PR DESCRIPTION
**Pull request by miguelgfierro**
### Description

for rating: predict
for ranking: recommend_top_k
also applied black

### Checklist:

- [ ] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [ ] This PR is being made to `staging` and not `master`.

### CODE REVIEW

File: reco_utils/recommender/surprise/surprise_utils.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def compute_rating_predictions(
     return predictions.drop(["details", "r_ui"], axis="columns")
 
 
-def compute_ranking_predictions(
+def recommend_k_items(
```
Comment (anargyri) [inline reco_utils/recommender/surprise/surprise_utils.py]: This function does not recommend k items. It computes the predictions for all users and items.

Comment (miguelgfierro) [reply reco_utils/recommender/surprise/surprise_utils.py]: ohh good catch, the cut of k is done in fact when the metric is computed https://github.com/microsoft/recommenders/blob/master/notebooks/02_model/surprise_svd_deep_dive.ipynb

Comment (miguelgfierro) [reply reco_utils/recommender/surprise/surprise_utils.py]: also check with @yueguoguo whether or not we are using the threshold cut https://github.com/microsoft/recommenders/issues/1041

Comment (gramhagen) [reply reco_utils/recommender/surprise/surprise_utils.py]: if we don't want to work with the scores as a matrix like sar does we could use a heap

```
from heapq import heappush, heappushpop

users = data[usercol].unique()
items = data[itemcol].unique()
preds_lst = np.zeros([len(users), k])

for user_idx, user in enumerate(users):
    heap = []
    for item in items:
        score = algo.predict(user, item).est
        if len(heap) < k:
            heappush(heap, score)
        elif score > heap[k - 1]:
            heappushpop(heap, score);
    preds_lst[user_idx] = heap[::-1]
```

Comment (anargyri) [reply reco_utils/recommender/surprise/surprise_utils.py]: `heapq` also has an `nlargest()` method. Not sure which way would be faster (depends on which algorithm it uses).

Comment (gramhagen) [reply reco_utils/recommender/surprise/surprise_utils.py]: true, but it wasn't clear to me if that method tries to sort the items (which would be unnecessary) or leverages the heap?

Comment (anargyri) [reply reco_utils/recommender/surprise/surprise_utils.py]: https://github.com/python/cpython/blob/61b3484cdf27ceca1c1069a351487d2db4b2b48c/Lib/heapq.py#L395
It looks similar to your for loop.

**Comment by miguelgfierro** [comment]
hey guys @anargyri @gramhagen, I reverted the interface of ranking in surprise to do the work on a different PR. The related issue is here: https://github.com/microsoft/recommenders/issues/1042, please take a look once the tests have passed

**Pull request merged**